crates/apollo-router-core/src/response.rs

```diff
@@ -122,10 +130,12 @@ fn select_object(
         match selection {
             Selection::Field(field) => {
                 if let Some(value) = select_field(content, field, schema)? {
-                    output
-                        .entry(field.name.to_owned())
-                        .and_modify(|existing| existing.deep_merge(&value))
-                        .or_insert(value);
+                    match output.entry(field.name.to_owned()) {
+                        Entry::Occupied(mut existing) => existing.get_mut().deep_merge(value),
+                        Entry::Vacant(vacant) => {
+                            vacant.insert(value);
+                        }
+                    }
```

Comment:
```suggestion
                                        output
                        .entry(field.name.to_owned())
                        .or_insert_with(|| Default::default())
                        .deep_merge(value);
```

Reply:
I'm not sure it's a good idea to add an empty value to merge into when we can directly put the value we have

Reply:
not super sure bout that one perf wise 😅